tests/ssg_test_suite/common.py

```diff
@@ -308,3 +325,61 @@ def iterate_over_rules():
                 directory=tests_dir, id=full_rule_id, short_id=short_rule_id,
                 files=scripts)
             yield result
+
+
+def get_cpe_of_tested_os(domain_ip, logfile_name):
+    os_release_file = "/etc/os-release"
+    ret, cpe_line = run_cmd_remote(
+            "grep CPE_NAME {os_release_file}".format(os_release_file=os_release_file),
+            domain_ip, logfile_name)
+    # We are parsing an assignment that is possibly quoted
+    cpe = re.match(r'''CPE_NAME=(["']?)(.*)\1''', cpe_line)
+    if cpe and cpe.groups()[1]:
+        return cpe.groups()[1]
+    msg = ["Unable to get a CPE of the system running tests"]
+    if cpe_line:
+        msg.append(
+                "Retreived a CPE line that we couldn't parse: {cpe_line}"
+                .format(cpe_line=cpe_line))
+    else:
+        msg.append(
+                "Couldn't get CPE entry from '{os_release_file}'"
+                .format(os_release_file=os_release_file))
+    raise RuntimeError("\n".join(msg))
+
+
+INSTALL_COMMANDS = dict(
+        fedora=("dnf", "install", "-y"),
+        rhel7=("yum", "install", "-y"),
+        rhel8=("yum", "install", "-y"),
+)
+
+
+def install_packages(domain_ip, packages):
+    machine = "{0}@{1}".format(REMOTE_USER, domain_ip)
+    log_file_name = os.path.join(LogHelper.LOG_DIR, "env-preparation.log")
+
+    platform_cpe = get_cpe_of_tested_os(domain_ip, log_file_name)
+    platform = cpes_to_platform([platform_cpe])
+
+    with open(log_file_name, 'a') as log_file:
+        print("Installing packages", file=log_file)
+        log_file.flush()
+        execute_remote_command(
+                machine, INSTALL_COMMANDS[platform] + tuple(packages), log_file,
+                "Couldn't install required packages {packages}".format(packages=packages))
+
+
+def cpes_to_platform(cpes):
+    for cpe in cpes:
```

Comment:
Why does it eat a list? In the only usage you artifically create a list with a single item.

Reply:
It used to work with a list of CPEs that were associated with a benchmark, and to basically find the OS one among them. We don't use that right now, but I think that it may come in handy.